### ISSUE TITLE
[Linux] Dota crashing just after launch with Kernel 6.9rc5

### ISSUE THREAD

**Issue by AngryPenguinPL** [ValveSoftware/Dota-2#2694]
#### Your system information

* System information from steam (`Steam` -> `Help` -> `System Information`) in a [gist](https://gist.github.com/AngryPenguinPL/924d0db25c9d95b7bec4525b3832903c) here is a runtime diagnostics [file](https://gist.github.com/AngryPenguinPL/bd316ab31ee657188980739d9856c989)

* Have you checked for system updates?: [Yes]
* Are you using the latest stable video driver available for your system? [Yes]
* Have you verified the game files?: [Yes]

#### Please describe your issue in as much detail as possible:
Dota works fine on stable kernels from 6.8 branch and with 6.9 up to rc4. When booting into kernel 6.9rc5 game dota crashing just after launch (when valve logo is showing), or somethimes just after main menu game is displayed. By crashing I mean game just hang/freeze for 1 min and then quit to desktop.

In journal log i see this:
```
kwi 24 17:45:31 openmandriva-cooker systemd-coredump[28118]: [🡕] Process 28022 (dota2) of user 1001 dumped core.
                                                             
                                                             Stack trace of thread 28022:
                                                             #0  0x00007fd525c9b613 n/a (/run/host/usr/lib64/libvulkan_radeon.so + 0x69b613)
                                                             ELF object binary architecture: AMD x86-64
kwi 24 17:45:32 openmandriva-cooker systemd[1]: systemd-coredump@0-28117-0.service: Deactivated successfully.
kwi 24 17:45:32 openmandriva-cooker systemd[1]: systemd-coredump@0-28117-0.service: Consumed 17.916s CPU time.
kwi 24 17:45:32 openmandriva-cooker steam.desktop[28018]: /media/x/5afea239-9232-4445-a98e-4e892b1c4e26/steam/steamapps/common/dota 2 beta/game/dota.sh: line 122: 28022 Bus error               (core dumped) ${STEAM_RUNTIME_PREFIX} ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"
```
This is console log from steam launch up to game crash:
https://gist.github.com/AngryPenguinPL/4b0d712c1cc40402c7b0c2c9b438f4d1

I attaches also core dump from /tmp/dump below
[crash_20240424175348_2.dmp.txt](https://github.com/ValveSoftware/Dota-2/files/15097092/crash_20240424175348_2.dmp.txt)

To sum up, **dota2 crashes on kernel 6.9rc5, but on previous kernels like 6.9rc4 it works.**

#### Steps for reproducing this issue:

1. Install kernel 6.9rc5, boot into it
2. Run dota2
3. See crash.

**Comment by kisak-valve** [comment]
Hello @AngryPenguinPL, `libvulkan_radeon.so` hints that this is a regression in your video driver. You should report this issue to your video driver vendor.

**Comment by koreanfan** [comment]
archlinux 6.8.9-arch1-1, dota crash after launch:
Process 6194 (dota2) of user 1000 dumped core.                                                                  
Stack trace of thread 6260:
#0  0x0000700bed0fe633 n/a (/run/host/usr/lib/libvulkan_radeon.so + 0xfe633)
ELF object binary architecture: AMD x86-64
No problem on lts kernel

**Comment by koreanfan** [comment]
If you are lucky and dota dont crash immediately then go to video settings and set display mode to windowed and press apply. Kill dota or wait when dota crash. After that go to dota settings and delete all launch options. Launch dota. Seems no crash. If after those manipulation dota crash then try to kill dota and kill steam. Relaunch steam and launch dota, it should work but not stable. You can play demo mode and when quit dota and try to relaunch then dota may crash randomly. In replay mode dota crash. So i dont recommend to launch online match. So wait for fix from devels(mesa/valve).

**Comment by AngryPenguinPL** [comment]
Bug is created against kernel DRM, it is bisected. Please feel free to add comments here's https://gitlab.freedesktop.org/drm/amd/-/issues/3343

**Comment by koreanfan** [comment]
archlinux lts kernel 6.6.30-1-lts, dota crash after launch:
systemd-coredump[3966]: [🡕] Process 3885 (dota2) of user 1000 dumpe dcore.                                                                   
Stack trace of thread 3913:
#0  0x00007456bbcfe633 n/a (/run/host/usr/lib/libvulkan_radeon.so + 0xfe633)
ELF object binary architecture: AMD x86-64

**Comment by mfed3** [comment]
Confirmed seeing the same issue on Dota 2, both with the info in the gists and also after downgrading kernel to 6.6.30 still the same issue.
Operating System: NixOS 24.05
KDE Plasma Version: 6.0.4
Kernel Version: 6.8.9 (64-bit)
Graphics Platform: Wayland
Processors: 32 × AMD Ryzen 9 5950X 16-Core Processor
Memory: 62.7 GiB of RAM
Graphics Processor: AMD Radeon RX 7900 XT
System information from steam: https://gist.github.com/mfed3/213559ef909b232171158e273e92a1c1

Steam Runtime Diagnostics: https://gist.github.com/mfed3/6e16d24f4565dee225fc3c3a0b23753c

Crash Log: https://gist.github.com/mfed3/128116ff35c5bfbe08c59cc46a7754c1

**Comment by baaannnaaannn** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2694#issuecomment-2097387683

Me too...maybe they have some actual patches and applied to LTS kernel too?
I use 6.8.0 without any problem

**Comment by baaannnaaannn** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2694#issuecomment-2097387683

Found a solution?

**Comment by baaannnaaannn** [comment]
> To resolve the issue, downgrade to kernel 6.8.8.
> 
> Kernel 6.8.9 is an issue with the AMD GPU.

So its mesa driver bug or kernel?

**Comment by baaannnaaannn** [comment]
> Bug is created against kernel DRM, it is bisected. Please feel free to add comments here's https://gitlab.freedesktop.org/drm/amd/-/issues/3343

Its mesa driver bug or kernel?

**Comment by Zesko** [comment]
> Its mesa driver bug or kernel?

Kernels 6.8.9 and 6.6.30 have this bug. They should be fixed later.

**Comment by baaannnaaannn** [comment]
> > Its mesa driver bug or kernel?
> 
> Kernels 6.8.9 and 6.6.30 have this bug. They should be fixed later.

1.Does mesa driver coming with kernel,including kernel?(im new to Linux sry idk)
2.how can this effect into LTS kernel?

**Comment by Zesko** [comment]
1. No
2. Kernel 6.8.9 and 6.6.30 LTS have **the latest** path-versions at the same time, they get the same change of amdgpu driver. 

If you have "Above 4G decoding" in BIOS setting, enable it to resolve the issue. Or downgrade back to the previous path-version of the Kernel.

**Comment by baaannnaaannn** [comment]
> 1. No
> 2. Kernel 6.8.9 and 6.6.30 LTS have **the latest** path-versions at the same time, they get the same change of amdgpu driver. 
> 
> If you have "Above 4G decoding" in BIOS setting, enable it to resolve the issue. Or downgrade back to the previous path-version of the Kernel.

1.then if its driver issue why most of people in gitlab talking about kernel version not mesa driver?
Again sry for this,I just want to understand

**Comment by Zesko** [comment]
> 1.then if its driver issue why most of people in gitlab talking about kernel version not mesa driver?

Yes, the bug is in the amdgpu driver in the current kernel, not in Mesa. Mesa is not a amdgpu driver itself, but an implementation of OpenGL, Vulkan and other graphics APIs.

**Comment by baaannnaaannn** [comment]
> > 1.then if its driver issue why most of people in gitlab talking about kernel version not mesa driver?
> 
> Yes, the bug is in the amdgpu driver in the current kernel, not in Mesa. Mesa is not a amdgpu driver itself, but an implementation of OpenGL, Vulkan and other graphics APIs.

I just realized THANK YOU SO MUCH

**Comment by Thorinori** [comment]
> > Its mesa driver bug or kernel?
> 
> Kernels 6.8.9 and 6.6.30 have this bug. They should be fixed later.

Any idea when? Tried the Above 4G BIOS option and it worked once but then Dota just kept crashing still. Would ideally prefer to not have to roll back to an older kernel version, but definitely not a fan of the crashing.

**Comment by Zesko** [comment]
> Any idea when? 

Which distro do you use? 

Did you enable "Resize bar" in the BIOS setting?   If it does not help, there is something wrong with your system. 

Or build your own kernel with some bugfix paths. If you don't want to build it, try the bleeding edge kernel 6.9rc7

**Comment by Thorinori** [comment]
> > Any idea when?
> 
> Which distro do you use?
> 
> Did you enable "Resize bar" in the BIOS setting? If it does not help, there is something wrong with your system.
> 
> Or build your own kernel with some bugfix paths. If you don't want to build it, try the bleeding edge kernel 6.9rc7

EndeavourOS (Arch Based)

I don't recall seeing that, I did enable Above 4G though. Might have missed something though. I ended up just rolling back my kernel to 6.6.29 instead for now

**Comment by say8hi** [comment]
OS: Arch
WM: Hyprland
Kernel: 6.8.9-arch1-2
GPU: RX 6950 XT

Can downgrading amdgpu drivers help? And is it even possible?

**Comment by mfed3** [comment]
Honestly best thing to do is just enable Above 4G decoding in BIOS

**Comment by lewisdiamond** [comment]
> Honestly best thing to do is just enable Above 4G decoding in BIOS

I tried that with no success. Looks like rollback is necessary.

**Comment by say8hi** [comment]
> Honestly best thing to do is just enable Above 4G decoding in BIOS

Worked for me, ty

**Comment by lewisdiamond** [comment]
> > Honestly best thing to do is just enable Above 4G decoding in BIOS
> 
> I tried that with no success. Looks like rollback is necessary.

Turns out CSM has to be disabled for `Resize bar` to work. With `Above 4G decoding`=`enabled`, `Resize bar`=`auto` and `CSM`=`disabled` the problem is fixed.

**Comment by bemug** [comment]
Enabled `Above 4G decoding` and `Resizeable bar` and everything is back to normal. Kernel 6.8.9-arch1-2. Thank you.

**Comment by baaannnaaannn** [comment]
> Enabled `Above 4G decoding` and `Resizeable bar` and everything is back to normal. Kernel 6.8.9-arch1-2. Thank you.

What if my motherboard doesn't support 4G decoding and resize bar but problems still here
# I mean its not a solution

**Comment by lewisdiamond** [comment]
> I mean its not a solution

It's not. But it's not a Valve issue, it's a `amdgpu` bug which is tracked and being worked on. https://gitlab.freedesktop.org/drm/amd/-/issues/3343

> Christian König
[@ckoenig](https://gitlab.freedesktop.org/ckoenig) · [2 days ago](https://gitlab.freedesktop.org/drm/amd/-/issues/3343#note_2405355)
Owner
>And Alex pushed it into our internal branch, should be upstream by the end of the week.

The patch is here: https://patchwork.freedesktop.org/patch/593130/

**Comment by AngryPenguinPL** [comment]
Kernel 6.9 was released it contains fix for this bug.
So people who are affected by this issue can update their systems to 6.9 stable or alternatively wait for fixes in current branches like 6.8.10 and 6.6.31 - which should appear this week.

I think we can close this topic to avoid spam.
